tests/test_hf_api.py

```diff
@@ -4106,6 +4106,24 @@ def test_user_following(self) -> None:
         assert len(list(following)) > 500
 
 
+class DailyPaperApiTest(unittest.TestCase):
+    @with_production_testing
+    def test_papers_by_date(self) -> None:
+        papers = HfApi().list_papers(date="2024-09-18")
+        assert len(list(papers)) > 1
+
+    @with_production_testing
+    def test_papers_by_query(self) -> None:
+        papers = HfApi().list_papers(query="attention")
+        assert len(list(papers)) > 0
+
+    @with_production_testing
+    def test_get_paper_by_id(self) -> None:
+        paper_id = "1706.03762"
+        paper = HfApi().paper_info(paper_id)
+        assert paper.title == "Attention Is All You Need"
```

Comment:
```suggestion
class PaperApiTest(unittest.TestCase):
    @classmethod
    @with_production_testing
    def setUpClass(cls) -> None:
        cls.api = HfApi()
        return super().setUpClass()

    def test_papers_by_query(self) -> None:
        papers = list(self.api.list_papers(query="llama"))
        assert len(papers) > 0
        assert "The Llama 3 Herd of Models" in [paper.title for paper in papers]

    def test_get_paper_by_id_success(self) -> None:
        paper = self.api.paper_info("2407.21783")
        assert paper.title == "The Llama 3 Herd of Models"
```

Something like this works for me (added `setUpClass`  + removed "filter by date" test + added explicit test for query)

Reply:
**EDIT:** let's also add a new test to check what happens if the paper does not exist:

```py
    def test_get_paper_by_id_not_found(self) -> None:
        with self.assertRaises(HfHubHTTPError) as context:
            self.api.paper_info("1234.56789")
        assert context.exception.response.status_code == 404
```